### PR TITLE
[fix] Properly define app upgradability / Fix app part of tools update

### PR DESCRIPTION
**Pull request by alexAubin**
This is a continuation of "clean app upgrade" and more refactoring of app.py in general. This PR defines an "upgradable" property directly in app_info (when used with the mysterious --raw option), which in turn allows to fix the related bug https://dev.yunohost.org/issues/787 in tools.py.

### CODE REVIEW

**Comment by alexAubin** [comment]
Bump :confused:

**Comment by alexAubin** [comment]
:c

File: src/yunohost/app.py [hunk 1]
```diff
@@ -380,18 +396,18 @@ def app_upgrade(auth, app=[], url=None, file=None):
...
-        elif 'lastUpdate' not in app_dict or 'git' not in app_dict:
+        elif app_dict["upgradable"] == "url_required":
             logger.warning(m18n.n('custom_app_url_required', app=app_instance_name))
             continue
-        elif app_dict['lastUpdate'] > locale_update_time:
+        elif app_dict["upgradable"] == "yes":
             manifest, extracted_app_folder = _fetch_app_from_git(app_instance_name)
         else:
+            if user_specified_list:
+                logger.success(m18n.n('app_already_up_to_date', app=app_instance_name))
```
Comment (zamentur) [inline src/yunohost/app.py]: We might could add a warning for apps that hasn't got update since 1 year ?

File: src/yunohost/tools.py [hunk 2]
```diff
@@ -357,37 +356,27 @@ def tools_update(ignore_apps=False, ignore_packages=False):
...
+        logger.info(m18n.n('done'))
 
+    # "apps" will list upgradable packages
     apps = []
     if not ignore_apps:
         try:
             app_fetchlist()
         except MoulinetteError:
+            # FIXME : silent exception !?
             pass
```
Comment (zamentur) [inline src/yunohost/tools.py]: It has been written 3 years ago
https://github.com/YunoHost/yunohost/commit/5335ffbcbef07e1f815b5f05ca700cc8746c5a4a

File: src/yunohost/tools.py [hunk 3]
```diff
@@ -357,37 +356,27 @@ def tools_update(ignore_apps=False, ignore_packages=False):
...
-                    original_app_id = app_id[:app_id.index('__')]
-                else:
-                    original_app_id = app_id
-
-                current_app_dict = app_info(app_id,  raw=True)
-                new_app_dict     = app_info(original_app_id, raw=True)
-
-                # Custom app
-                if new_app_dict is None or 'lastUpdate' not in new_app_dict or 'git' not in new_app_dict:
-                    continue
```
Comment (zamentur) [inline src/yunohost/tools.py]: SO it fix https://dev.yunohost.org/issues/322
 :+1:

**Pull request merged**